firefox-ios/Client/Frontend/Browser/SearchEngines/Views/SearchEngineSelectionViewController.swift

```diff
@@ -8,10 +8,60 @@ import UIKit
 import Shared
```

Comment:
This file basically serves as the "interface" for one example usage of the generic classes in BrowserKit.

As mentioned there are a few other options to consider... e.g. inheritance.

If we used inheritance we could override any of the tableview properties we wish. If we use generics, we have to decide between a delegate pattern or passing in functions with some sort of view model so our concrete implementations are flexible enough.

We could also consider some mix of the two. 🤔  Maybe generic data types in a concrete general table view where subclasses can call super to use default implementations and override UITableView delegate/dataSource methods where desired. (Actually, the more I think about this, the more I think it bears some investigation to see if it looks a bit cleaner than this example).

Just some more thoughts I had this AM cc @adudenamedruby